ivy_models/mlpmixer/mlpmixer.py

```diff
@@ -0,0 +1,111 @@
+        self.gelu = ivy.GELU()
+        self.linear1 = ivy.Linear(mlp_dim, inp_dim)
+        self.dropout = ivy.Dropout(0.25)
+        super(MLPBlock, self).__init__()
+
+    def _forward(self, x):
+        y = self.linear0(x)
+        y = self.gelu(y)
+        # y = self.dropout(y)
+        y = self.linear1(y)
+        # y = self.dropout(y)
+        return y
+
+
+class MixerBlock(ivy.Module):
+    def __init__(self, hidden_dim, num_patches, token_mlp_dim, channel_mlp_dim):
+        self.norm1 = ivy.LayerNorm([hidden_dim])
+        self.mlp_block1 = MLPBlock(num_patches, token_mlp_dim)
+        self.norm2 = ivy.LayerNorm([hidden_dim])
+        self.mlp_block2 = MLPBlock(hidden_dim, channel_mlp_dim)
+        super(MixerBlock, self).__init__()
+
+    def _forward(self, x):
+        y = self.norm1(x)
+        y = ivy.matrix_transpose(y)
+        y = self.mlp_block1(y)
+        y = ivy.matrix_transpose(y)
+        x = x + y
+        y = self.norm2(x)
+        y = x + self.mlp_block2(y)
+        return y
+
+
+class MLPMixer(ivy.Module):
+    def __init__(
+        self,
+        vol,
+        inp_dim,
+        num_classes=10,
+        num_blocks=2,
+        patch_size=16,
+        hidden_dim=128,
+        token_mlp_dim=64,
+        channel_mlp_dim=32,
+        v=None,
+    ):
+        self.num_blocks = num_blocks
+        self.conv = ivy.Conv2D(
+            inp_dim, hidden_dim, [patch_size, patch_size], [patch_size, patch_size], 0
+        )
+        self.num_patches = int((vol / patch_size) ** 2)
+        self.mixer_block = MixerBlock(
+            hidden_dim, self.num_patches, token_mlp_dim, channel_mlp_dim
+        )
+        self.norm = ivy.LayerNorm([hidden_dim])
+        self.dropout = ivy.Dropout(0.25)
+        self.linear = ivy.Linear(hidden_dim, num_classes)
+        super(MLPMixer, self).__init__(v=v)
+
+    def _forward(self, x):
+        x = self.conv(x)
+        x = x.reshape(
+            (int(x.shape[0]), int(x.shape[1]) * int(x.shape[2]), int(x.shape[3]))
+        )
+        for i in range(self.num_blocks):
+            x = self.mixer_block(x)
+        x = self.norm(x)
+        # x = self.dropout(x)
+        x = ivy.mean(x, axis=1)
+        logits = self.linear(x)
+        probs = ivy.softmax(logits)
+
+        return probs
+
+
+def mlpmixer(
+    pretrained=True,
+    vol=72,
+    inp_dim=3,
+    num_classes=10,
+    num_blocks=4,
+    patch_size=9,
+    hidden_dim=128,
+    token_mlp_dim=64,
+    channel_mlp_dim=128,
+):
+    """Ivy MLPMixer model"""
+    model = MLPMixer(
+        vol,
+        inp_dim,
+        num_classes=num_classes,
+        num_blocks=num_blocks,
+        patch_size=patch_size,
+        hidden_dim=hidden_dim,
+        token_mlp_dim=token_mlp_dim,
+        channel_mlp_dim=channel_mlp_dim,
+    )
+    if pretrained:
+        REPO_ID = "unifyai/MLPMixer"
+        FILENAME = "weights.pickled"
+        weights = joblib.load(hf_hub_download(repo_id=REPO_ID, filename=FILENAME))
```

Comment:
once the `BaseModel` class is inherited, we have a built-in load method, where you can use `MLPMixer.load_from_huggingface(repo_id)` and it'll return the loaded model object, so it'll be something like 
```
if pretrained:
    return MLPMixer.load_from_huggingface(repo_id)
return model = MLPMixer(
         vol,
         inp_dim,
         num_classes=num_classes,
         num_blocks=num_blocks,
         patch_size=patch_size,
         hidden_dim=hidden_dim,
         token_mlp_dim=token_mlp_dim,
         channel_mlp_dim=channel_mlp_dim,
     )
```